keel-intents/src/main/kotlin/com/netflix/spinnaker/keel/intents/processors/ApplicationIntentProcessor.kt

```diff
@@ -79,8 +83,51 @@ class ApplicationIntentProcessor
   }
 
   private fun currentStateUpToDate(currentState: Application?, desiredState: BaseApplicationSpec): Boolean {
-    log.warn("Current state update check is not implemented")
-    return false
+    log.info("Determining if state has drifted for application ${desiredState.name}")
+    if (currentState == null) return false
+
+    // Some values are computed, and we shouldn't consider them when
+    // determining if the current state is equal to the desired state
+    val ignoredCurrentStateParameters: MutableList<String> = mutableListOf()
+    Application::class.primaryConstructor!!.parameters.filter { param ->
+        param.findAnnotation<Computed>()?.ignore == true
+      }.mapTo(ignoredCurrentStateParameters) { param ->
+        param.name.toString()
+      }
+    ignoredCurrentStateParameters.addAll(currentState.computedPropertiesToIgnore())
+
+    val ignoredDesiredStateParameters: MutableList<String> = mutableListOf()
+    val specPrimaryConstructor = NetflixApplicationSpec::class.primaryConstructor!!
```

Comment:
What if Google comes along with their own extension of `BaseApplicationSpec`? Is there some way we can abstract this behind something like... `fun <T : BaseApplicationSpec> getPrimarySpecConstructor<T>(): T`? (I think that syntax is slightly wrong).

At any rate, good step. We'll probably want to find a way to make all of this abstracted so we can use it across all of the intent processors.

Reply:
Ah forgot to abstract that part. Good suggestion, will address in the future, have added a great TODO. Merging for now so we stop bombing orca with tasks.